### ISSUE TITLE
How can I setup the minecraft console clients on a raspberry pi?

### ISSUE THREAD

**Issue by AnthonyWoolley** [ORelio/Minecraft-Console-Client#373]
How can I setup 8 minecraft console clients on a raspberry pi.

**Comment by CarbonNeuron** [comment]
You can run it using mono! I think the command is "sudo mono (file)"
However, from my experience- running 2 clients makes the raspberry pi really hot and have to be cooled by putting it on my air vent

**Comment by ORelio** [comment]
You do not need to run mono as administrator (so it is `mono MinecraftClient.exe` not `sudo mono MinecraftClient.exe`). For more information about how to install mono see (README)[https://github.com/ORelio/Minecraft-Console-Client/blob/master/MinecraftClient/config/README.txt#L19]. Also you may use the [`screen`](https://www.digitalocean.com/community/tutorials/how-to-install-and-use-screen-on-an-ubuntu-cloud-server) program to run several instances.

**Comment by lansraad** [comment]
How can I download the MinecraftClient on my Raspberry Pi? Iv managed to download mono but im not sure how to get the actual file onto my pi

Im on Rasperian stretch on a Raspberry pi 2 B

**Comment by alicenchains** [comment]
Use FileZilla or Core FTP. Put in the hostname and connect to it, transfer files over to the root. I personally use Core FTP although it seems to mess up s lot.

Sent from my iPhone

On Nov 11, 2018, at 11:15 AM, lansraad <notifications@github.com<mailto:notifications@github.com>> wrote:


How can I download the MinecraftClient on my Raspberry Pi? Iv managed to download mono but im not sure how to get the actual file onto my pi

Im on Rasperian stretch on a Raspberry pi 2 B

**Comment by ZizzyDizzyMC** [comment]
As Rasperian Stretch follows a Debian package system, the same installation steps used for Ubuntu / Debian should work in this case.  
This is a guide on how to do this via a terminal assuming you're using another desktop to connect to the pi.  You can do it on the pi itself too, if you open up a terminal window and a browser to visit the website linked below.
Create a directory you want to have the client with `mkdir <name>`
Navigate to directory with `cd <name>`
Get the latest development release from Appveyor, https://ci.appveyor.com/project/ORelio/minecraft-console-client/build/artifacts .  From that page download the exe and move it to the directory you created.  Alternately right click the link, copy link address.  Go to the terminal (where you are already in the directory) and type `wget ` and then paste the link in place to download via wget.  
Once the file is downloaded, run the client with `mono MinecraftClient.exe`  


To answer the original question, you can open multiple clients at once via the program `screen` on linux, if you don't have it, `sudo apt-get install screen` on a debian based system.

The following would open up multiple virtual terminals with their own client running within.
```
screen -dmS username1 mono MinecraftClient.exe <your-INI-file.ini> <username1> <password1>
screen -dmS username2 mono MinecraftClient.exe <your-INI-file.ini> <username2> <password2>
screen -dmS username3 mono MinecraftClient.exe <your-INI-file.ini> <username3> <password3>
```

**Comment by TheSnoozer** [comment]
Thanks for taking the time to answer the question :-)
Since this seems to be answered I guess I can close it. feel free to reopen if you still have any issues in getting this sorted. Otherwise please open a new issue.

**Comment by TheHapsPro** [comment]
> You can run it using mono! I think the command is "sudo mono (file)"
> However, from my experience- running 2 clients makes the raspberry pi really hot and have to be cooled by putting it on my air vent

I didnt have the problem of overheating. I ran up to 6 clients at the same time and the temperature were around 50 degrees. I have the raspberry pi 3 and I kept it running for 2 weeks and the temperature didnt increase. 
I did have a issue at first tho, `Unsupported protocol version 578` , but it was a easy fix just put in `MinecraftClient.ini` `version=1.8.9`. 
I would also recomend using `screen`, because its much easier to start the clients.
